diff --git a/Makefile.common b/Makefile.common
index dbafa0b..f55dbc0 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -51,11 +51,11 @@ AVR_INC 	:= ${AVR_ROOT}/avr-4/
 AVR 		:= ${AVR_ROOT}/bin/avr-
 # Thats for MacPorts libelf
 ifeq (${shell test -d /opt/local && echo Exists}, Exists)
-IPATH		+= /opt/local/include
-LFLAGS		= -L/opt/local/lib/
+IPATH		+= /opt/local/include:/opt/local/include/libelf
+LFLAGS		= -L/opt/local/lib/ -L${AVR_ROOT}/lib
 endif
 else
-LINUX_AVR_ROOTS := /usr/lib/avr /usr/avr /opt/cross/avr/avr /usr/local/avr
+LINUX_AVR_ROOTS := /usr/lib/avr /usr/avr /opt/cross/avr/avr /usr/local/avr /opt/local/avr
 AVR_ROOT := $(firstword $(wildcard $(LINUX_AVR_ROOTS)))
 ifeq (${AVR_ROOT},)
 $(error No avr-libc root directory found. Tried the following paths: $(LINUX_AVR_ROOTS))
diff --git a/simavr/Makefile b/simavr/Makefile
index aebc070..f822c15 100644
--- a/simavr/Makefile
+++ b/simavr/Makefile
@@ -43,7 +43,7 @@ sim_o 	:= ${patsubst sim/%.c, ${OBJ}/%.o, ${sim}}
 VPATH	= cores
 VPATH	+= sim
 
-IPATH	= sim
+IPATH	+= sim
 IPATH	+= .
 IPATH	+= ../../shared
     
@@ -188,7 +188,7 @@ sim_core_decl.h: sim_core_config.h $(cores) Makefile
 	for core in $$(grep -r avr_kind_t cores/|awk -F '[ :]' '{print $$1 "=" $$3;}') ; do \
 		file=$${core/=*}; global=$${core/*=}; \
 		upper=$${file/cores\/sim_}; upper=$${upper/.c}; \
-		upper=$$(echo $$upper|tr '[a-z]' '[A-Z]'); \
+		upper=$$(echo $$global|tr '[a-z]' '[A-Z]'); \
 		decl+="#if CONFIG_$$upper\nextern avr_kind_t $$global;\n#endif\n"; \
 		array+="#if CONFIG_$$upper\n\t&$$global,\n#endif\n"; \
 	done ; \
